awx/main/utils/redis.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2025 Ansible, Inc.
+# All Rights Reserved
+
+"""Redis client utilities with automatic retry on connection errors."""
+
+import redis
+import redis.asyncio
+from django.conf import settings
+from redis.backoff import ExponentialBackoff
+from redis.retry import Retry
+from redis.exceptions import BusyLoadingError, ConnectionError, TimeoutError
+
+
+def _get_redis_pool_kwargs():
+    """
+    Get common Redis connection pool kwargs with retry configuration.
+
+    Returns:
+        dict: Keyword arguments for redis.ConnectionPool.from_url()
+    """
+    retry = Retry(ExponentialBackoff(cap=1.0, base=0.5), retries=settings.REDIS_RETRY_COUNT)
```

Comment:
I would still suggest putting `cap` and `base` into settings, so they can be adjusted in a worst-case scenario.